torchao/quantization/quantize_/workflows/int8/int8_tensor.py

```diff
@@ -0,0 +1,302 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD 3-Clause license found in the
+# LICENSE file in the root directory of this source tree.
+
+from dataclasses import dataclass
+from typing import List, Optional
+
+import torch
+from torch.utils._python_dispatch import return_and_correct_aliasing
+
+from torchao.float8.inference import _slice_scale_for_dimension
+from torchao.kernel import int_scaled_matmul
+from torchao.quantization.granularity import Granularity, PerRow, PerTensor
+from torchao.quantization.quant_primitives import (
+    MappingType,
+    choose_qparams_affine,
+    dequantize_affine,
+    quantize_affine,
+)
+from torchao.quantization.quantize_.common import QuantizeTensorKwargs
+from torchao.quantization.utils import get_block_size
+from torchao.utils import TorchAOBaseTensor, fill_defaults
+
+__all__ = ["Int8Tensor", "QuantizeTensorToInt8Kwargs"]
+
+aten = torch.ops.aten
+
+
+@dataclass
+class QuantizeTensorToInt8Kwargs(QuantizeTensorKwargs):
+    """Tensor kwargs for creating int8 tensor from high precision
+
+    Args:
+        granularity: the granularity for the Tensor, currently either PerRow() or PerTensor()
+    """
+
+    granularity: Granularity = PerRow()
+
+
+class Int8Tensor(TorchAOBaseTensor):
+    """
+    int8 quantized tensor with plain layout
+
+    Tensor Attributes:
+        qdata: (N, K) or (B, N, K) int8 quantized weight data (2D or 3D)
+        scale: scale factors for dequantization
+        # TODO: Static quantization support using `static_scale`
+
+    Non-Tensor Attributes:
+        granularity: the granularity for quantization (e.g., PerRow(), PerTensor())
+        act_quant_kwargs: flags for dynamic activation quantization
+    """
+
+    # TODO: Static quantization support using `static_scale`
+    tensor_data_names = ["qdata", "scale"]
+    tensor_attribute_names = []
+    optional_tensor_attribute_names = [
+        "block_size",
+        "act_quant_kwargs",
+        "dtype",
```

Comment:
nit: should dtype be optional? or just required?

Oh I guess this is following float8 tensor, and only used for dequantize for now. I think one thing to make sure if just it still works when `dtype` == None